main.nf

```diff
@@ -13,47 +13,93 @@ nextflow.enable.dsl = 2
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    GENOME PARAMETER VALUES
+    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
+include { VARIANTMTB                } from './workflows/variantmtb'
+include { PIPELINE_INITIALISATION   } from './subworkflows/local/utils_nfcore_variantmtb_pipeline'
+include { PIPELINE_COMPLETION       } from './subworkflows/local/utils_nfcore_variantmtb_pipeline'
+
+include { getGenomeAttribute        } from './subworkflows/local/utils_nfcore_variantmtb_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    GENOME PARAMETER VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-WorkflowMain.initialise(workflow, params, log)
+// if specified, fetch fasta file from --genome parameter
+params.fasta = getGenomeAttribute('fasta')
```

Comment:
can we do this in the workflow maybe?

Reply:
the tamplate does it here, intuitively I would have moved it to `PIPELINE_INITIALIZATION()`. workflow also sounds good to me.